tests/catalog/test_sql.py

```diff
@@ -168,6 +168,39 @@ def test_creation_with_unsupported_uri(catalog_name: str) -> None:
         SqlCatalog(catalog_name, uri="unsupported:xxx")
 
 
+@pytest.mark.parametrize(
+    "echo_param, expected_echo_value",
+    [(None, strtobool(DEFAULT_ECHO_VALUE)), ("debug", "debug"), ("true", True), ("false", False)],
```

Comment:
nit: can we move these parameterized values into the test itself as a for-loop? this can cut down number of tests and thus the overall test time

Reply:
I can certainly do that, but I typically find that for-loops are often hiding on which input the test is failing.  You need to provide custom assert messages or mentally figure out which input is triggering the failure.  With parameters, pytest will tell you right away which one is failing.  But I understand the overhead provided by parameterized test.  Should I still go ahead with the requested change?

Reply:
my preference is to use for-loop instead of parameterized tests.

> for-loops are often hiding on which input the test is failing

I've seen patterns like 
```
    assert catalog.engine._echo == expected_echo_value, (
        f"Assertion failed: expected echo value {expected_echo_value}, "
        f"but got {catalog.engine._echo}. For echo_param={echo_param}"
    )
```
which makes it easier to figure out which variable inside the for-loop is causing the assertion error